MITLibraries/browsertrix-harvester
# Align btrix CLI arguments for v0.12.0 release

### What does this PR do?

Aligns example crawl config YAML to have new syntax for `text: to-pages` (formerly just `text: true`).

### Helpful background context

Browsertrix had a [minor release v0.12.0](https://github.com/webrecorder/browsertrix-crawler/releases/tag/v0.12.0) that changed the optional fulltext CLI argument `--text`, now explicitly defining where the fulltext should be stored (where previously it defaulted to pages JSON files).

New browsertrix CLI parameters for fulltext:
```text
--text                                Extract initial (default) or final t
                                            ext to pages.jsonl or WARC resource
                                            record(s)
                       [array] [choices: "to-pages", "to-warc", "final-to-warc"]
```

In time, we may want to leverage storing fulltext in WARC which is now possible from this change, as better for performance and storage, but okay to continue looking for fulltext in pages JSON files.

### How can a reviewer verify these changes?

```shell
make dist-local
make run-harvest-local
```

Confirm that crawl is successful, and output metadata records at `output/crawls/collections/homepage/homepage.xml` have keywords present, which indicates fulltext was saved during crawl and extracted during metadata records creation.

Also, can confirm new browsetrix version running is `v0.12.0`:
```text
2023-11-03 13:33:53,409 DEBUG harvester.crawl.crawl() line 81: {"timestamp":"2023-11-03T13:33:53.408Z","logLevel":"info","context":"general","message":"Browsertrix-Crawler 0.12.0 (with warcio.js 1.6.2 pywb 2.7.4)","details":{}}
```

### Includes new or updated dependencies?

YES

### What are the relevant tickets?

None

### Developer

- [X] All new ENV is documented in README (or there is none)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes
